app.py

```diff
@@ -317,11 +316,24 @@ def homepage():
     """
 
     if g.user:
-        messages = (Message
-                    .query
-                    .order_by(Message.timestamp.desc())
-                    .limit(100)
-                    .all())
+        following = g.user.following
+        new_messages = []
+        for i in range(len(following)):
+            for f in following[i].messages:
+                new_messages.append(f)
+
+        all_messages = (Message
+            .query
+            .order_by(Message.timestamp.desc())
+            .all())
```

Comment:
Again let's not pull all rows into memory like this. You should be able to filter by message ids here using new messages